server/src/routes/v2/apps.js

```diff
@@ -19,27 +28,28 @@ module.exports = [
                 query: Joi.object({
                     channels: Joi.filter(
                         Joi.stringArray().items(Joi.valid(...CHANNELS))
-                    ).description(
-                        'Filter by channel'
-                    ).default(['stable']),
+                    )
+                        .description('Filter by channel')
+                        .default(['stable']),
                     types: Joi.filter(
                         Joi.stringArray().items(Joi.valid(...APPTYPES))
-                    ).description(
-                        'Filter by app type'
-                    ).default(['APP']),
+                    )
+                        .description('Filter by app type')
+                        .default(['APP']),
                 }).unknown(true),
             },
             plugins: {
                 queryFilter: {
-                    enabled: true
+                    enabled: true,
                 },
                 pagination: {
                     enabled: true,
                 },
             },
         },
         handler: async (request, h) => {
-            const channels = request.plugins.queryFilter.getFilter('channels').value
+            const channels = request.plugins.queryFilter.getFilter('channels')
```

Comment:
Overall it seems like this is implemented like the old v1 API-structure, which we wanted to move away from. There's some abstractions that are used for the Organisation that I would like to see be used more, and see how it performs in a more complex case like this. It may be that the abstractions are not fit for it, as Organiastion are very simple compared to this. But I would like to see how it worked out. 

Maybe we should try to do a refactor after, as the primary focus for this PR is not the `v2`-API. I do think that the main changes you've done could be done directly in the `v1` API, as that needs fixing for the user-problem either way. I also don't think we should delete it.
We need to rethink the `v2`-API either way, as we want to be able to create apps without uploading an app at the same time as well. So in my mind the `v2`-API should warrant more changes than this - not just API-facing but overall code structure.

Reply:
>  There's some abstractions that are used for the Organisation that I would like to see be used more, and see how it performs in a more complex case like this

I've tried to base the create handler on [the create organisation handler](https://github.com/dhis2/app-hub/blob/0075932ebae37d852599c77f69e5d50385a817e6/server/src/routes/v2/organisations.js#L124-L143) so I'm not sure what other abstractions used for organisations I should be using but feel free to push the changes you're thinking of to this PR.

I've tried to limit the backend changes I've made in this PR due to i) the large diff, and ii) the focus being on the frontend - hence my original update to the create app handler being pretty much copy-paste from the v1 with only the changes that I required for the frontend to work.

> I do think that the main changes you've done could be done directly in the v1 API, as that needs fixing for the user-problem either way. I also don't think we should delete it.

I'll move the v2 create app handler (as well as the tests) back to v1 :+1: